addons/l10n_ar/models/account_move.py

```diff
@@ -34,6 +34,25 @@ def _l10n_ar_get_document_number_parts(self, document_number, document_type_code
     l10n_ar_afip_service_start = fields.Date(string='AFIP Service Start Date', readonly=True, states={'draft': [('readonly', False)]})
     l10n_ar_afip_service_end = fields.Date(string='AFIP Service End Date', readonly=True, states={'draft': [('readonly', False)]})
 
+    def _is_manual_document_number(self):
+        """ Document number should be manual input by user when the journal use documents and
+
+        * if sales journal and not a AFIP pos (liquido producto case)
+        * if purchase journal and not a AFIP pos (regular case of vendor bills)
+
+        All the other cases the number should be automatic set, wiht only one exception, for pre-printed/online AFIP
+        POS type, the first numeber will be always set manually by the user and then will be computed automatically
+        from there """
+        if self.country_code != 'AR':
+            return super()._is_manual_document_number()
+
+        # NOTE: There is a corner case where 2 sales documents can have the same number for the same DOC from a
+        # different vendor, in that case, the user can create a new Sales Liquido Producto Journal
+        return True if self.l10n_latam_use_documents and (
+            (self.journal_id.type == 'purchase' and not self.journal_id.l10n_ar_is_pos) or
+            (self.journal_id.type == 'sale' and not self.journal_id.l10n_ar_is_pos)
+        ) else False
```

Comment:
```suggestion
        return self.l10n_latam_use_documents and (
            (self.journal_id.type == 'purchase' and not self.journal_id.l10n_ar_is_pos) or
            (self.journal_id.type == 'sale' and not self.journal_id.l10n_ar_is_pos)
        ) 
```

Reply:
@zaoral ping

Reply:
push the last version with the simpler if, and remove the f/else condition

Reply:
add unit tests

Reply:
we simplify the method as we mentioned in the last call. and also added unit tests